src/main/scala/algolia/definitions/WaitForTaskDefinition.scala

```diff
@@ -64,6 +60,19 @@ trait WaitForTaskDsl {
   implicit object WaitForTaskDefinitionExecutable
       extends Executable[WaitForTaskDefinition, TaskStatus] {
 
+    // Run every 50 ms, use a wheel with 512 buckets
+    private lazy val timer = {
+      val threadFactory = new ThreadFactory {
```

Comment:
It will always be one thread. The real configuration is what `HashedWheelTimer` offers as parameters:

 - `ThreadFactory` to determine how to create this single worker thread
 - the `tickDuration`: how often are timed out tasks being executed by the worker thread
 - the size of the wheel

I am not sure users care too much about all of them but maybe the `tickDuration` if they are willing to go for "more often" or "more batching". I guess we can add a `tickDuration` as a `FiniteDuration` in the `WaitForTaskDefinition`. 

Something like:

    case class WaitForTaskDefinition(taskId: Long,
                                 index: Option[String] = None,
                                 baseDelay: Long = 100,
                                 maxDelay: Long = Long.MaxValue,
                                 tickDuration: Long = 50)  // using Long here for consistency with the other fields
    ...

    new HashedWheelTimer(threadFactory, query.tickDuration, TimeUnit.MILLISECONDS, 512)